### PR TITLE
Change: Update view permissions on Environment Variables

### LINKED ISSUE #3404: Allow suitably credentialed users (developers?) to view the names of existing environment variables
Currently, the RBAC matrix denies permission to "view" certain environment variables (project-scope, production env).

Ideally, non-maintainers who need to know these variables exist should be able to do so easily. Should we create a new permission - View Environment Variable Name (or similar) and grant that to all developers under the current RBAC.

This would allow them to at least know they exist, even if they aren't sure of the value.

### PR DESCRIPTION
**Pull request by CGoodwin90**
Addresses #3404 

- Updates the existing policies for viewing environment variables to cover guest level access. 
- Creates new scopes for viewing environment variable values -   `environment:viewValue:development`, `environment:viewValue:production` & `project:viewValue`
- Updates the resolver to check if value is requested and perform relevent permission checks

# Checklist

- [x] Affected Issues have been mentioned in the Closing issues section
- [x] Documentation has been written/updated
- [ ] PR title is ready for inclusion in changelog

# Closing issues

closes #3404

### CODE REVIEW

**Comment by tobybellwood** [comment]
sidenote: We'll need to reflect these new permissions in the :scream: RBAC matrix in the docs...

**Comment by CGoodwin90** [comment]
> sidenote: We'll need to reflect these new permissions in the scream RBAC matrix in the docs...

Yep added them in - also tried to update the RBAC matrix to make it more readable, let me know if you think it works

File: services/keycloak/startup-scripts/00-configure-lagoon.sh [hunk 1]
```diff
@@ -1959,6 +1959,110 @@ EOF
...
+  view_value_env_var_dev_environment=$(/opt/jboss/keycloak/bin/kcadm.sh get -r lagoon clients/$CLIENT_ID/authz/resource-server/permission?name=View+Environment+Variable+Value+for+Development+Environment --config $CONFIG_PATH | jq -r '.[0]["id"]')
+  view_value_env_var_prod_environment=$(/opt/jboss/keycloak/bin/kcadm.sh get -r lagoon clients/$CLIENT_ID/authz/resource-server/permission?name=View+Environment+Variable+Value+for+Production+Environment --config $CONFIG_PATH | jq -r '.[0]["id"]')
+  view_value_env_var_project=$(/opt/jboss/keycloak/bin/kcadm.sh get -r lagoon clients/$CLIENT_ID/authz/resource-server/permission?name=View+Environment+Variable+Value+for+Project --config $CONFIG_PATH | jq -r '.[0]["id"]')
+
+  if [ "$view_value_env_var_dev_environment" != null ]; then
+      echo "environment:viewValue:development on env_var already configured"
+      return 0
+  fi
+
+  echo Recreating \"View Environment Variable for Development Environment\" permissions
```
Comment (shreddedbacon) [inline services/keycloak/startup-scripts/00-configure-lagoon.sh]: Should move these echo statements from here to above the sections that the events take place, just so the logs reflect what is happening as they happen.

File: services/api/src/resources/env-variables/resolvers.ts
Comment (shreddedbacon) [inline services/api/src/resources/env-variables/resolvers.ts:1]: Can you fix up the indentations in this file

**Review by shreddedbacon — CHANGES_REQUESTED** [review]
Only minor indentation fixes requested, once they're done all good

**Review by shreddedbacon — APPROVED** [review]

**Pull request merged**